### ISSUE TITLE
WrongExpectedVersion when deleting a projection with all delete options

### ISSUE THREAD

**Issue by hayley-jean** [EventStore/EventStore#3005]
Deleting a projection with all delete options checked results in a `WrongExpectedVersion` error and other errors printed in the server logs. The delete request itself reports that it was successful.

## Reproduction Steps:

1. Start an empty Event Store with projections enabled.
2. Create a simple projection. Leave all options default (this fails for both Continuous and OneTime projections):

```
fromStream("test-stream")
.when({
    $init: function() {
        return { count: 0 };
    },
    $any: function(s,e) {
        s.count++;
    }
})
```

3. Delete the projection through the UI, checking all of the delete options: 

![image](https://user-images.githubusercontent.com/5140165/123058111-c3848180-d408-11eb-9cbc-f95b82fb4abf.png)

4. Check the server logs. The following errors should be logged:

```
[15940,16,09:47:20.984,INF] PROJECTIONS: Projection Stream '"$projections-demo-emittedstreams-checkpoint"' deleted
[15940,16,09:47:20.986,INF] PROJECTIONS: Projection '"demo"' deletion has not been written to "$projections-$all". Error: "WrongExpectedVersion"
[15940,16,09:47:20.986,ERR] PROJECTIONS: Got wrong expected version writing projection deletion for "demo".
[15940,16,09:47:20.986,INF] PROJECTIONS: Projection Stream '"$projections-demo-emittedstreams"' deleted
[15940,16,09:47:20.986,ERR] Error while processing message "EventStore.Projections.Core.Messages.ProjectionManagementMessage+Internal+Deleted" in queued handler '"Projections Leader"'.
System.NullReferenceException: Object reference not set to an instance of an object.
   at EventStore.Projections.Core.Services.Management.ProjectionManager.Handle(Deleted message) in C:\Users\hayle\code\EventStore\src\EventStore.Projections.Core\Services\Management\ProjectionManager.cs:line 616
   at EventStore.Core.Bus.MessageHandler`1.TryHandle(Message message) in C:\Users\hayle\code\EventStore\src\EventStore.Core\Bus\MessageHandler.cs:line 30
   at EventStore.Core.Bus.InMemoryBus.Publish(Message message) in C:\Users\hayle\code\EventStore\src\EventStore.Core\Bus\InMemoryBus.cs:line 290
   at EventStore.Core.Bus.QueuedHandlerMRES.ReadFromQueue(Object o) in C:\Users\hayle\code\EventStore\src\EventStore.Core\Bus\QueuedHandlerMRES.cs:line 119
```

## Event Store Details:

Version: 21.6
Projection Runtime: Legacy and Interpreted

**Comment by StevenBlair123** [comment]
@hayley-jean - I had a quick look at this for you, and it looks like the associated streams are being deleted.
Do you simply want a null check in _Handle(ProjectionManagementMessage.Internal.Deleted message)_ to prevent the Eventstore from falling over?

**Comment by hayley-jean** [comment]
@StevenBlair123 Thanks! That sounds like the correct fix if you're willing to submit a PR 😄

**Comment by oskardudycz** [comment]
This was fixed by #3014.
